src/module-elasticsuite-core/etc/di.xml

```diff
@@ -408,4 +408,19 @@
     <type name="Smile\ElasticsuiteCore\Index\Mapping\Field">
         <plugin name="mappingBackwardCompatibility" type="Smile\ElasticsuiteCore\Plugin\Deprecation\Index\Mapping\FieldPlugin"/>
     </type>
+
+    <!-- Sensitive / Environment Specific Configuration fields handling -->
+    <type name="Magento\Config\Model\Config\TypePool">
+        <arguments>
+            <argument name="environment" xsi:type="array">
+                <item name="smile_elasticsuite_core_base_settings/es_client/servers" xsi:type="string">1</item>
+                <item name="smile_elasticsuite_core_base_settings/es_client/enable_https_mode" xsi:type="string">1</item>
+                <item name="smile_elasticsuite_core_base_settings/es_client/enable_http_auth" xsi:type="string">1</item>
+            </argument>
+            <argument name="sensitive" xsi:type="array">
+                <item name="smile_elasticsuite_core_base_settings/es_client/http_auth_user" xsi:type="string">1</item>
```

Comment:
Should we add http_auth_user and http_auth_pwd to the environment variables too ?

Reply:
Yes, I was thinking the same when doing this PR, but I was confused by Magento not systematically using it when I did search for other usages of environment/sensitive settings inside legacy modules.

I will amend this PR.